conan/tools/meson/toolchain.py

```diff
@@ -296,13 +296,8 @@ def _resolve_apple_flags_and_variables(self, build_env, compilers_by_conf):
                 "Apple SDK path not found. For cross-compilation, you must "
                 "provide a valid SDK path in 'tools.apple:sdk_path' config."
             )
-
-        # TODO: Delete this os_sdk check whenever the _guess_apple_sdk_name() function disappears
-        os_sdk = self._conanfile.settings.get_safe('os.sdk')
-        if not os_sdk and self._os != "Macos":
-            raise ConanException("Please, specify a suitable value for os.sdk.")
-
         # Calculating the main Apple flags
+        os_sdk = get_apple_sdk_fullname(self._conanfile)
```

Comment:
Using this reads weird:
- The ``get_apple_sdk_fullname`` returns a value with the sdk+sdk_version.
- But then, later, this value is passed to ``apple_min_version_flag()`` which requires the sdk_version separately, ignores the value in the ``os_sdk`` and use the ``os.version`` instead.

The whole flow is confusing (it is not because of this PR, it was already confusing)

Reply:
Yes, I agree. Do you want me to open an issue to solve this?

Reply:
I just saw that this flow also appears in the AutotoolsToolchain generator.

Reply:
I see, good point. Still readability and understanding what and why it is doing it is complicated. Lets try to do a later refactor of this then.